api/src/main/java/io/grpc/InternalRetryTracer.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2021 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+/** API for tracing retry events. */
```

Comment:
I'm not seeing anything in the retry code that guarantees `shouldRetry()` isn't called after `commit()`, and it seems the two methods could be called concurrently. That's probably fine for our intended usage, but we should call it out here.

Reply:
Call out with `@ThreadSafe`. Implementation must guarantee thread-safety of its state.